src/cclib/parser/daltonparser.py

```diff
@@ -764,9 +770,9 @@ def extract(self, inputfile, line):
         # ...
         #
         if "Final HF energy" in line and not (hasattr(self, "mpenergies") or hasattr(self, "ccenergies")):
-            self.metadata["theory"] = "HF"
+            self.metadata["methods"] = "HF"
         if "Final DFT energy" in line:
-            self.metadata["theory"] = "DFT"
+            self.metadata["methods"] = "DFT"
```

Comment:
The metadata has an item called "functional" to be used for exacting the name of the DFT functional. Once the DFT data is published on our portal, it could identify this functional is GGA or hybrid one, or it has dispersion correction or not.